lib/Desugar.hs

```diff
@@ -31,6 +31,41 @@ import Data.List        (intercalate)
 import Text.PrettyPrint.ANSI.Leijen
 
 import qualified Data.Map as Map
+import qualified Data.Set as Set (member)
+
+
+-- | Generate a fresh variable in the context. This is a very general API and
+-- requires caution from the user, that is, it is the responsibility of the user
+-- to ensure that the fresh variable is inserted to VarMap in the ensuing code.
+-- For example,
+--
+--     `Map.insert x (F.Var x x') g`
+-- where x' is a bound *Haskell* variable.
+--
+-- Example:
+-- If you want a fresh variable at the position denoted by "?" in the following
+-- program, ``\x. \y. \?. e` with the current `VarMap` being `g`, then you can
+-- write `let x = generateFreshVar g in ...` so that inside `...` you have
+-- the fresh variable `x` as well as the updated `VarMap` `g1`.
```

Comment:
Note that at the last line, `g1` is a new name (not `g`). How to make a `g1` is mentioned in the doc above the example.